### PR TITLE
sql: split bootstrap out of tenant create

### PR DESCRIPTION
**Pull request by azhu-crl**
Release note: none.
Epic: none.

### CODE REVIEW

**Review by dt — APPROVED** [review]

**Comment by princejha95** [comment]
Added few minor comments

**Comment by azhu-crl** [comment]
@princejha95 Sorry I don't think your comments were posted?

**Comment by princejha95** [comment]
> @princejha95 Sorry I don't think your comments were posted?

Sorry, my bad. Missed to publish the comments. Thanks for informing.

**Review by stevendanna — APPROVED** [review]
LGTM. Left a couple of style nits that probably aren't worth another CI run.

File: pkg/sql/tenant_creation.go [hunk 1]
```diff
@@ -140,9 +141,21 @@ func (p *planner) createTenantInternal(
 		return tid, nil
 	}
 
+	return BootstrapTenant(ctx, p.execCfg, p.Txn(), info, initialTenantZoneConfig)
+}
+
+// BootstrapTenant bootstraps the span of the newly created tenant identified in
+// the passed tenant info using the passed zone config.
+func BootstrapTenant(
```
[the remark this answers is not in the document]
Comment (msbutler) [reply pkg/sql/tenant_creation.go:149]: I think it's fine to merge this without tests. This is a pure refactor PR, and this code is already well tested via higher level tenant creation tests.

File: pkg/sql/tenant_creation.go [hunk 2]
```diff
@@ -140,9 +141,21 @@ func (p *planner) createTenantInternal(
...
+	ctx context.Context,
+	execCfg *ExecutorConfig,
+	txn *kv.Txn,
+	info mtinfopb.TenantInfoWithUsage,
+	zfcg *zonepb.ZoneConfig,
+) (roachpb.TenantID, error) {
```
[the remark this answers is not in the document]
Comment (azhu-crl) [reply pkg/sql/tenant_creation.go:155]: There's one usage of `TenantID` in `crdb_internal.create_tenant` builtin so changing the return values of `BootstrapTenant` would involve further refactoring. Is it necessary to remove it?

File: pkg/sql/tenant_creation.go [hunk 3]
```diff
@@ -140,9 +141,21 @@ func (p *planner) createTenantInternal(
...
 	}
 
+	return BootstrapTenant(ctx, p.execCfg, p.Txn(), info, initialTenantZoneConfig)
+}
+
+// BootstrapTenant bootstraps the span of the newly created tenant identified in
+// the passed tenant info using the passed zone config.
+func BootstrapTenant(
+	ctx context.Context,
+	execCfg *ExecutorConfig,
```
[the remark this answers is not in the document]
Comment (azhu-crl) [reply pkg/sql/tenant_creation.go:151]: Just wanted to clarify, are you referring to `BootstrapTenant()` or another function? The only usage of `BootstrapTenant()` is on line 144

File: pkg/sql/tenant_creation.go [hunk 4]
```diff
@@ -140,9 +141,21 @@ func (p *planner) createTenantInternal(
...
+	txn *kv.Txn,
+	info mtinfopb.TenantInfoWithUsage,
+	zfcg *zonepb.ZoneConfig,
+) (roachpb.TenantID, error) {
 	// Retrieve the possibly auto-generated ID.
```
[the remark this answers is not in the document]
Comment (azhu-crl) [reply pkg/sql/tenant_creation.go]: Removed it in the update

**Comment by azhu-crl** [comment]
As discussed with @msbutler offline, we'll merge this pr to unblock PCR work.

bors r+

**Pull request merged**